code/game/objects/machinery/cloning/cloning.dm

```diff
@@ -0,0 +1,237 @@
+
+	var/obj/machinery/cloning/vats/linked_machine
+
+
+/obj/machinery/cloning_console/vats/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	if(user.a_intent != INTENT_HELP)
+		return
+
+	if(!linked_machine)
+		// Try to find the machine nearby
+		linked_machine = locate() in range(1)
+		if(!linked_machine)
+			visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'connection not available'.</span>")
+			return TRUE
+
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps as its boots up and connects to \the [linked_machine]</span>")
+		return TRUE
+
+	if(linked_machine.occupant || linked_machine.timerid)
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'already processing clone'.</span>")
+		return TRUE
+
+	if(!linked_machine.beaker || linked_machine.beaker.reagents.total_volume < linked_machine.biomass_required)
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'not enough biomass'.</span>")
+		return TRUE
+
+	visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> whirls as it starts to create a new clone.</span>")
+	linked_machine.grow_human()
+
+
+/obj/machinery/cloning/vats
+	name = "Clone Vat"
+	icon = 'icons/obj/machines/cloning.dmi'
+	icon_state = "cell_100"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 900
+
+	var/timerid
+	var/mob/living/carbon/human/occupant
+	var/obj/item/reagent_containers/glass/beaker
+
+	var/biomass_required = 40
+	var/grow_timer = 15 MINUTES
+
+
+/obj/machinery/cloning/vats/Initialize()
+	. = ..()
+	beaker = new /obj/item/reagent_containers/glass/beaker/biomass
+
+
+/obj/machinery/cloning/vats/Destroy()
+	deltimer(timerid)
+	timerid = null
+
+	// Force tthe clone out, if they have a client
+	if(occupant?.client)
+		eject_user()
+
+	QDEL_NULL(occupant)
+	QDEL_NULL(beaker)
+	return ..()
+
+
+/obj/machinery/cloning/vats/relaymove(mob/user)
+	eject_user()
+	return FALSE
+
+
+/obj/machinery/cloning/vats/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	if(user.a_intent == INTENT_HARM)
+		user.visible_message("<span class='notice'>[src] bangs on the glass</span>", "<span class='notice'>You bang on the glass</span>")
+		return TRUE
+
+	if(!beaker)
+		return
+
+	if(timerid || occupant) // You need to stop the process or remove the human first.
+		to_chat(user, "<span class='notice'>You can't get to the beaker while the machine growing a clone.</span>")
+		return
+
+	beaker.forceMove(drop_location())
+	user.put_in_hands(beaker)
+	beaker = null
+
+	update_icon()
+
+	return TRUE
+
+
+/obj/machinery/cloning/vats/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(.)
+		return
+
+	if(!ispath(I, /obj/item/reagent_containers/glass/beaker))
+		if(beaker)
+			to_chat(user, "<span class='warning'>A beaker is already loaded into the machine.</span>")
+			return
```

Comment:
Did you mean `if(ispath)` here?

Reply:
I did but in testing ispath wasn't working at all for me or hugh

Reply:
Oh, this should be:
```suggestion
	if(istype(I, /obj/item/reagent_containers/glass/beaker))
		if(beaker)
			to_chat(user, "<span class='warning'>A beaker is already loaded into the machine.</span>")
			return
```
`ispath` is to check if a variable is a type path variable

Reply:
Oh sorry no i was looking at the wrong lines and read what you said wrong,
You're right i need istype.